### PR TITLE
Add test cases for `try` statement

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR adds test cases for the `try` statement.

### CODE REVIEW

File: crates/ruff_python_parser/src/parser/statement.rs [hunk 1]
```diff
@@ -1181,27 +1183,158 @@ impl<'src> Parser<'src> {
...
+            } else {
+                // test_err except_stmt_missing_as_name
+                // try:
+                //     pass
+                // except Exception as:
+                //     pass
+                // except Exception as
+                //     pass
+                self.add_error(
+                    ParseErrorType::OtherError("Expected symbol name after `as`".to_string()),
```
Comment (MichaReiser) [inline crates/ruff_python_parser/src/parser/statement.rs:1314]: Nit: As a user it wouldn't be clear what symbol refers to. The documentation refers to the part after `as` as the target, but I think that might still be hard to understand. Maybe just use `Expected name after as` or `Expected variable name after `as`?

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@except_stmt_missing_exception.py.snap [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        ],
+    },
+)
+```
+## Errors
+
+  |
+1 | try:
+2 |     pass
+3 | except as exc:
```
Comment (MichaReiser) [inline crates/ruff_python_parser/tests/snapshots/invalid_syntax@except_stmt_missing_exception.py.snap:121]: Nit: What happens if you have `except as:`

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_invalid_order.py.snap [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        ],
+    },
+)
+```
+## Errors
+
+  |
+3 | finally:
+4 |     pass
+5 | else:
```
Comment (MichaReiser) [inline crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_invalid_order.py.snap:48]: Do I understand this correctly that the `else` gets dropped as part of the statement error recovery code?

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_missing_except_finally.py.snap [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        ],
+    },
+)
+```
+## Errors
+
+  |
+1 | try:
+2 | try:
+  | ^^^ Syntax Error: Expected an indented block after `try` statement
```
Comment (MichaReiser) [inline crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_missing_except_finally.py.snap:47]: I don't think this is the intended error message. Any chance that the second `try:` clause is unintentional?

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@except_stmt_missing_exception.py.snap (hunk 2 above)
Comment (dhruvmanila) [reply crates/ruff_python_parser/tests/snapshots/invalid_syntax@except_stmt_missing_exception.py.snap:121]: Two error messages as expected. I'll add this case, thanks!

```
Errors:
-------
Syntax Error: Expected one or more exception types at byte range 21..23
  |
1 | try:
2 |     pass
3 | except as:
  |        ^^ Syntax Error: Expected one or more exception types
4 |     pass
  |


Syntax Error: Expected name after `as` at byte range 23..24
  |
1 | try:
2 |     pass
3 | except as:
  |          ^ Syntax Error: Expected name after `as`
4 |     pass
  |

```

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_invalid_order.py.snap (hunk 3 above)
Comment (dhruvmanila) [reply crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_invalid_order.py.snap:48]: Hmm, it won't be dropped but there will be additional error as then it's outside the `try` statement. So, the statements in the `else` body will be parsed as if it's outside the `try` statement. I think it should be trivial to recover from this by parsing it again.

File: crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_missing_except_finally.py.snap (hunk 4 above)
Comment (dhruvmanila) [reply crates/ruff_python_parser/tests/snapshots/invalid_syntax@try_stmt_missing_except_finally.py.snap:47]: Yeah, that was unintentional. Thanks!

**Pull request merged**